research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,306 @@
+import os
+import sys
+import time
+import shutil
+import zipfile
+import urllib.request
+
+import pandas as pd
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+DATA_PATH = os.path.join(BASE_DIR, 'data')
+
+
+def reporthook(blocknum, blocksize, totalsize):
+    readsofar = blocknum * blocksize
+    if totalsize > 0:
+        percent = readsofar * 1e2 / totalsize
+        s = "\r%5.1f%% %*d / %d" % (
+            percent, len(str(totalsize)), readsofar, totalsize)
+        sys.stderr.write(s)
+        if readsofar >= totalsize:
+            sys.stderr.write("\n")
+    else:
+        sys.stderr.write("read %d\n" % (readsofar,))
+
+
+def folder_walk_2010():
```

Comment:
Hi @lacerdamarcelo I have some suggestions of refactoring to your code. I was talking to @cuducos this morning and one thing that we can refactor is here, we have four methods `def folder_walk_YYYY()` what do you think about creating only one method and then use the year as a parameter. They may change a bit, but they almost look quite the same, so we can refactor that :)